hack/update-vendor.sh

```diff
@@ -72,28 +72,54 @@ function ensure_require_replace_directives_for_all_dependencies() {
   # collect 'replace' directives that unconditionally pin versions (old=new@version)
   local replace_filter='(.Old.Version == null) and (.New.Version != null)'
 
-  # Capture local require/replace directives before running any go commands that can modify the go.mod file
+  # Capture local require/replace directives from root and staging modules
+  # before running any go commands that can modify the go.mod file
   local require_json="${local_tmp_dir}/require.json"
   local replace_json="${local_tmp_dir}/replace.json"
-  go mod edit -json \
-      | jq -r ".Require // [] | sort | .[] | select(${require_filter})" \
-      > "${require_json}"
-  go mod edit -json \
-      | jq -r ".Replace // [] | sort | .[] | select(${replace_filter})" \
-      > "${replace_json}"
 
-  # Propagate root replace/require directives into staging modules, in case we are downgrading, so they don't bump the root required version back up
+  collect_require_replace_directives "${require_json}" "${replace_json}"
   for repo in $(kube::util::list_staging_repos); do
     pushd "staging/src/k8s.io/${repo}" >/dev/null 2>&1
-      jq -r '"-require \(.Path)@\(.Version)"' < "${require_json}" \
-          | xargs -L 100 go mod edit -fmt
-      jq -r '"-replace \(.Old.Path)=\(.New.Path)@\(.New.Version)"' < "${replace_json}" \
-          | xargs -L 100 go mod edit -fmt
+    collect_require_replace_directives "${require_json}" "${replace_json}"
```

Comment:
we don't want to collect `replace` directives from staging repos (they shouldn't have any that differ from the root anyway)

maybe split to `collect_require_directives` and `collect_replace_directives`